### PR TITLE
Refactor runner script

### PR DESCRIPTION
**Pull request by tiziano88**
The data types are much clearer now, and it's getting closer to being
able to support async steps via Tokio (in a future PR).

Add support for timing individual commands.

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [x] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by ipetr0v** [comment]
Why do we need async steps in `runner` if they are already provided by `cloudbuild`?

File: runner/src/main.rs [hunk 1]
```diff
@@ -125,125 +124,153 @@ fn is_cargo_workspace_file(path: &PathBuf) -> bool {
...
+                        "--",
+                        "--deny=warnings",
+                    ],
+                ),
+            })
+            .collect(),
+    }
+}
+
+fn run_bazel_build() -> Step {
```
Comment (ipetr0v) [inline runner/src/main.rs]: Should it be called `run_bazel_build_all`?

File: runner/src/main.rs [hunk 2]
```diff
@@ -33,9 +33,9 @@ use internal::*;
 fn main() -> Result<(), Box<dyn std::error::Error>> {
```
Comment (ipetr0v) [inline runner/src/main.rs]: If it possible to run a server or examples using the `runner`, or it only runs Cargo related tasks and Bazel builds?

Comment (tiziano88) [reply runner/src/main.rs]: I am going to add that functionality in the next PR.

**Comment by tiziano88** [comment]
> Why do we need async steps in `runner` if they are already provided by `cloudbuild`?

- cloudbuild parallelism is very coarse grain (only scripts can be run in parallel in different steps)
- it is not easy to run locally the same steps that cloudbuild runs, apart from looking at `.cloudbuild.yaml` and trying to run them in the same order; `runner` is supposed to work both locally and in cloudbuild

File: runner/src/main.rs (hunk 1 above)
Comment (tiziano88) [reply runner/src/main.rs]: I'll rename later when I add more precise rules anyways.

**Pull request merged**